packages/reference/src/assets/WrappedAssetCard/FetchingWrappedAssetCard.tsx

```diff
@@ -73,23 +77,42 @@ export function FetchingWrappedAssetCard(props: FetchingWrappedAssetCardProps) {
             slide,
           });
       },
-      cardDragHandle: props.cardDragHandle,
       onRemove: onRemoveAsset,
     };
 
+    function renderDefaultCard(props?: CustomCardProps) {
+      return <WrappedAssetCard size={size} {...commonProps} {...props} />;
+    }
+
     if (props.viewType === 'link') {
       if (asset === undefined) {
         return <EntryCard size="small" loading />;
       }
       return <WrappedAssetLink {...commonProps} />;
     }
 
-    const size = props.viewType === 'big_card' ? 'default' : 'small';
-
     if (asset === undefined) {
       return <AssetCard size={size} isLoading title="" src="" href="" />;
     }
 
-    return <WrappedAssetCard size={size} {...commonProps} />;
+    if (props.renderCustomCard) {
+      const customProps = {
+        ...commonProps,
+        size: size as 'default' | 'small',
+      };
+
+      // LinkActionsProps are injected higher SingleReferenceEditor/MultipleReferenceEditor
+      const renderedCustomCard = props.renderCustomCard(
+        customProps,
+        {} as LinkActionsProps,
+        renderDefaultCard
+      );
+      // Only `false` indicates to render the original card. E.g. `null` would result in no card.
+      if (renderedCustomCard !== false) {
```

Comment:
What is the use case of having a renderCustomCard that returns false? If we want the default card, shouldn't we not provide a renderCustomCard at all?

Reply:
We better ask @DanweDE. But I think the idea is that you can decide inside the custom card renderer where you have access to all the computed props, specifically loaded `asset`. For example you could say that an asset of "published" status should be rendered as default card, otherwise as custom.

Reply:
Yes, that's the idea here :point_up: you can basically make the decision whether or not to render the custom card dynamically at "run time" and for each card individually rather than just during configuration of the component.